packages/react-tabs/src/components/Tab/Tab.types.ts

```diff
@@ -39,19 +54,28 @@ export type TabProps = ComponentProps<Partial<TabSlots>> & TabCommons;
  * State used in rendering Tab
  */
 export type TabState = ComponentState<TabSlots> &
-  TabCommons & {
+  Omit<TabCommons, 'disabled'> &
+  Required<Pick<TabCommons, 'disabled'>> & {
     /**
      * A tab supports 'transparent' and 'subtle' appearance.
      */
     appearance?: string;
     /**
-     * If this tab is selected.
+     * A tab can have only an icon slot filled and no content.
+     */
+    iconOnly: boolean;
+    /**
+     * The current selected tab value.
      */
-    selected?: boolean;
+    selectedValue?: TabValue;
```

Comment:
Is there need to hold `selectedValue` in the state? Should it be just `selected`?

Reply:
The value is the key into the registered tabs which allows the tab to find the selected and previously selected refs.  This is necessary for the animation approach design decided on.

Reply:
I see... Current implementation is below:

```ts
const selectedValue = useContextSelector(TabListContext, ctx => ctx.selectedValue);
```

`selectedValue` comes from `props` on `TabList`, correct? With current design:

```
// TabList: renders, new value for `selectedValue` is passed to context
//   Tab1: gets context update, `selectedValue` updated => re-render
//   Tab2: gets context update, `selectedValue` updated => re-render
//   TabX: gets context update, `selectedValue` updated => re-render
```

With `selected` (_is selected_):

```ts
const selected = useContextSelector(TabListContext, ctx => ctx.selectedValue === value);
```

⬇️⬇️⬇️

```
// TabList: renders, new value for `selectedValue` is passed to context
//   Tab1: gets context update, `selected` => `false` => re-render
//   Tab2: gets context update, `selected` => `true` => re-render
//   Tab2: gets context update, `selected` not changed => skipped
//   TabX: gets context update, `selected` not changed => skipped
```

---

I see current implementation as a performance issue. Do you have any ideas how it could be avoided?

Reply:
OK! I have updated the context selector and provided the tab the tab registration information via a callback. Verified with react dev tools that only the selected and previously selected tabs re-render when selectedValue changes.